puppy.py

```diff
@@ -28,4 +29,177 @@ def __init__(self, behavior_matches):
         # the puppy.
         self.touch_sensor = TouchSensor(Port.S1)
 
+    def walk_in_place(puppy, speed=100):
+        puppy.right_leg_motor.run_angle(speed, 40, wait=False)
+        puppy.left_leg_motor.run_angle(speed, -40, wait=True)
+        puppy.right_leg_motor.run_angle(speed, -40, wait=False)
+        puppy.left_leg_motor.run_angle(speed, 40, wait=True)
+        puppy.right_leg_motor.run_angle(speed, -40, wait=False)
+        puppy.left_leg_motor.run_angle(speed, 40, wait=True)
+        puppy.right_leg_motor.run_angle(speed, 40, wait=False)
+        puppy.left_leg_motor.run_angle(speed, -40, wait=True)
+
+    def sit(left_leg_motor, right_leg_motor):
+        left_leg_motor.run(-1000)
+        right_leg_motor.run(-1000)
+        wait(2000)
+        left_leg_motor.stop()
+        right_leg_motor.stop()
+
+    def hop(left_leg_motor, right_leg_motor):
+        left_leg_motor.run(500)
+        right_leg_motor.run(500)
+        wait(275)
+        left_leg_motor.hold()
+        right_leg_motor.hold()
+        wait(275)
+        left_leg_motor.run(-50)
+        right_leg_motor.run(-50)
+        wait(275)
+        left_leg_motor.stop()
+        right_leg_motor.stop()
+
+    def step_0_node_0_action(self): 
+        ev3.screen.load_image(NEUTRAL_EYES)
+
+    def step_0_node_1_action(self):
+        if self.behavior_matches:
+            head_motor.run(1000)
+            wait(1000)
+            self.ev3.speaker.say("uh oh")
+            head_motor.run(-1000)
+            wait(1000)
+        else:
+            pass
+  
+    def step_1_node_0_action(self):
+        if self.behavior_matches:
+            walk_in_place(self, 100)
```

Comment:
this should be self.walk_in_place(100). self does not go inside the parentheses. it's silly python syntax.